lib/jekyll_pages_api/generator.rb

```diff
@@ -10,7 +12,12 @@ def initialize(site)
     end
 
     def pages
-      self.site.pages.map{|page| Page.new(page) }.select(&:html?)
+      result = Array.new
+      self.site.each_site_file do |p|
+        p = Page.new(p, @site)
```

Comment:
Tsk tsk single-letter variables :wink: Especially this one.

http://ruby-doc.org/core-2.2.2/Kernel.html#method-i-p